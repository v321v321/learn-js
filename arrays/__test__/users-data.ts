export interface User {
  name: string;
  experience: number;
}

export const usersData = [
  { name: 'Zemlak', experience: 14 },
  { name: 'Upton', experience: 13 },
  { name: 'Wisoky', experience: 3 },
  { name: 'Balistreri', experience: 8 },
  { name: 'Braun', experience: 13 },
  { name: 'Walsh', experience: 9 },
  { name: 'Cummings', experience: 3 },
  { name: 'Johnston', experience: 8 },
  { name: 'Dickens', experience: 14 },
  { name: 'Treutel', experience: 6 },
  { name: 'Buckridge', experience: 12 },
  { name: 'Veum', experience: 13 },
  { name: 'Turcotte', experience: 10 },
  { name: 'Bartell', experience: 8 },
  { name: 'Gottlieb', experience: 7 },
  { name: 'Will', experience: 15 },
  { name: 'Gleason', experience: 14 },
  { name: 'Nikolaus', experience: 8 },
  { name: 'Durgan', experience: 4 },
  { name: 'Mayert', experience: 14 },
  { name: 'Flatley', experience: 8 },
  { name: 'Koelpin', experience: 3 },
  { name: 'Feest', experience: 14 },
  { name: 'Thiel', experience: 10 },
  { name: 'Collins', experience: 8 },
  { name: 'Hudson', experience: 6 },
  { name: 'Crona', experience: 7 },
  { name: 'Kihn', experience: 13 },
  { name: 'Murray', experience: 8 },
  { name: 'Rempel', experience: 2 },
  { name: 'Gleichner', experience: 6 },
  { name: 'Keebler', experience: 10 },
  { name: 'Abernathy', experience: 4 },
  { name: 'Wolf', experience: 4 },
  { name: 'Fadel', experience: 11 },
  { name: 'Simonis', experience: 15 },
  { name: 'Ullrich', experience: 7 },
  { name: 'Jones', experience: 5 },
  { name: 'Langworth', experience: 4 },
  { name: 'Lind', experience: 14 },
  { name: 'Kuphal', experience: 9 },
  { name: 'Jerde', experience: 4 },
  { name: 'Fritsch', experience: 4 },
  { name: 'Stanton', experience: 3 },
  { name: 'Turner', experience: 9 },
  { name: 'Nolan', experience: 9 },
  { name: 'Gerlach', experience: 3 },
  { name: 'Boyle', experience: 10 },
  { name: 'Pfannerstill', experience: 6 },
  { name: 'Cormier', experience: 9 },
  { name: 'Padberg', experience: 12 },
  { name: 'Purdy', experience: 13 },
  { name: 'Ritchie', experience: 4 },
  { name: 'Doyle', experience: 12 },
  { name: 'Reichert', experience: 13 },
  { name: 'Breitenberg', experience: 13 },
  { name: 'Erdman', experience: 11 },
  { name: 'Schumm', experience: 9 },
  { name: 'Pfeffer', experience: 3 },
  { name: 'VonRueden', experience: 5 },
  { name: 'Wisozk', experience: 10 },
  { name: 'Metz', experience: 8 },
  { name: 'Barton', experience: 10 },
  { name: 'Ondricka', experience: 9 },
  { name: 'Corkery', experience: 14 },
  { name: 'Okuneva', experience: 13 },
  { name: 'Brown', experience: 13 },
  { name: 'Jacobson', experience: 12 },
  { name: 'Hills', experience: 5 },
  { name: 'Lueilwitz', experience: 4 },
  { name: 'Ziemann', experience: 10 },
  { name: 'Beier', experience: 8 },
  { name: 'Lemke', experience: 5 },
  { name: 'Altenwerth', experience: 2 },
  { name: 'Keeling', experience: 7 },
  { name: 'Crist', experience: 4 },
  { name: 'Deckow', experience: 3 },
  { name: 'Pollich', experience: 6 },
  { name: 'Gusikowski', experience: 7 },
  { name: 'Stehr', experience: 5 },
  { name: 'Jast', experience: 9 },
  { name: 'Nicolas', experience: 4 },
  { name: 'Hermiston', experience: 2 },
  { name: 'Moore', experience: 12 },
  { name: 'Kautzer', experience: 2 },
  { name: 'Huel', experience: 15 },
  { name: 'Heidenreich', experience: 8 },
  { name: 'Conn', experience: 4 },
];
