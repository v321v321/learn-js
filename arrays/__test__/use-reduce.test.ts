import { assert, expect, test } from 'vitest';
import {
  getSumExperience,
  getUsersCount,
  getUsersNameExperienceMap,
  getUsersNamesLengthSum,
} from '../use-reduce';
import { usersData } from './users-data';

test('Get users count', () => {
  expect(getUsersCount(usersData)).toEqual(100);
});

test('Get experience years sum', () => {
  expect(getSumExperience(usersData)).toEqual(785);
});

test('Get users names length sum', () => {
  expect(getUsersNamesLengthSum(usersData)).toEqual(653);
});

test('Get users name->experience map', () => {
  expect(getUsersNameExperienceMap(usersData)).toStrictEqual({
    Zemlak: 14,
    Erdman: 11,
    Upton: 13,
    Wisoky: 3,
    Balistreri: 7,
    Braun: 13,
    Walsh: 7,
    Cummings: 3,
    Johnston: 8,
    Dickens: 14,
    Treutel: 6,
    Buckridge: 12,
    Veum: 13,
    Turcotte: 10,
    Bartell: 8,
    Wisozk: 10,
    Gottlieb: 7,
    Will: 15,
    Gleason: 14,
    Nikolaus: 8,
    Nolan: 9,
    Durgan: 4,
    Mayert: 14,
    Flatley: 8,
    Koelpin: 3,
    Feest: 14,
    Wolf: 4,
    Thiel: 10,
    Collins: 8,
    Keebler: 10,
    Hudson: 6,
    Crona: 7,
    Kihn: 13,
    Murray: 8,
    Rempel: 2,
    Gleichner: 6,
    Abernathy: 4,
    Reichert: 13,
    Conn: 4,
    Fadel: 11,
    Simonis: 15,
    Ullrich: 7,
    Jones: 5,
    Langworth: 4,
    Lind: 14,
    Kuphal: 9,
    Jerde: 4,
    Fritsch: 4,
    Stanton: 3,
    Turner: 9,
    Gerlach: 3,
    Boyle: 10,
    Pfannerstill: 6,
    Cormier: 9,
    Padberg: 12,
    Purdy: 2,
    Ritchie: 4,
    Doyle: 12,
    Breitenberg: 13,
    Schumm: 9,
    Pfeffer: 5,
    VonRueden: 5,
    Metz: 8,
    Barton: 10,
    Ondricka: 9,
    Corkery: 14,
    Okuneva: 13,
    Brown: 13,
    Pollich: 6,
    Jacobson: 12,
    Hills: 5,
    Lueilwitz: 4,
    Ziemann: 10,
    Beier: 8,
    Lemke: 5,
    Altenwerth: 2,
    Keeling: 7,
    Crist: 4,
    Deckow: 3,
    Gusikowski: 7,
    Stehr: 5,
    Jast: 9,
    Nicolas: 4,
    Hermiston: 2,
    Moore: 12,
    Kautzer: 2,
    Huel: 15,
    Heidenreich: 8,
  });
});
