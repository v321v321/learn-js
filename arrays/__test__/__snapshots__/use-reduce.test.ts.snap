// Vitest Snapshot v1

exports[`Get experience years sum 1`] = `727`;

exports[`Get users count 1`] = `88`;

exports[`Get users name->experience map 1`] = `
{
  "Abernathy": 4,
  "Altenwerth": 2,
  "Balistreri": 8,
  "Bartell": 8,
  "Barton": 10,
  "Beier": 8,
  "Boyle": 10,
  "Braun": 13,
  "Breitenberg": 13,
  "Brown": 13,
  "Buckridge": 12,
  "Collins": 8,
  "Conn": 4,
  "Corkery": 14,
  "Cormier": 9,
  "Crist": 4,
  "Crona": 7,
  "Cummings": 3,
  "Deckow": 3,
  "Dickens": 14,
  "Doyle": 12,
  "Durgan": 4,
  "Erdman": 11,
  "Fadel": 11,
  "Feest": 14,
  "Flatley": 8,
  "Fritsch": 4,
  "Gerlach": 3,
  "Gleason": 14,
  "Gleichner": 6,
  "Gottlieb": 7,
  "Gusikowski": 7,
  "Heidenreich": 8,
  "Hermiston": 2,
  "Hills": 5,
  "Hudson": 6,
  "Huel": 15,
  "Jacobson": 12,
  "Jast": 9,
  "Jerde": 4,
  "Johnston": 8,
  "Jones": 5,
  "Kautzer": 2,
  "Keebler": 10,
  "Keeling": 7,
  "Kihn": 13,
  "Koelpin": 3,
  "Kuphal": 9,
  "Langworth": 4,
  "Lemke": 5,
  "Lind": 14,
  "Lueilwitz": 4,
  "Mayert": 14,
  "Metz": 8,
  "Moore": 12,
  "Murray": 8,
  "Nicolas": 4,
  "Nikolaus": 8,
  "Nolan": 9,
  "Okuneva": 13,
  "Ondricka": 9,
  "Padberg": 12,
  "Pfannerstill": 6,
  "Pfeffer": 3,
  "Pollich": 6,
  "Purdy": 13,
  "Reichert": 13,
  "Rempel": 2,
  "Ritchie": 4,
  "Schumm": 9,
  "Simonis": 15,
  "Stanton": 3,
  "Stehr": 5,
  "Thiel": 10,
  "Treutel": 6,
  "Turcotte": 10,
  "Turner": 9,
  "Ullrich": 7,
  "Upton": 13,
  "Veum": 13,
  "VonRueden": 5,
  "Walsh": 9,
  "Will": 15,
  "Wisoky": 3,
  "Wisozk": 10,
  "Wolf": 4,
  "Zemlak": 14,
  "Ziemann": 10,
}
`;

exports[`Get users names length sum 1`] = `579`;
